x-pack/plugins/apm/public/components/shared/charts/TransactionCharts/index.tsx

```diff
@@ -8,220 +8,135 @@ import {
-                    </EuiTitle>
-                  </EuiFlexItem>
-                  <LicenseContext.Consumer>
-                    {(license) =>
-                      this.renderMLHeader(license?.getFeature('ml').isAvailable)
-                    }
-                  </LicenseContext.Consumer>
-                </EuiFlexGroup>
-                <TransactionLineChart
-                  series={responseTimeSeries}
-                  tickFormatY={getResponseTimeTickFormatter(formatter)}
-                  formatTooltipValue={getResponseTimeTooltipFormatter(
-                    formatter
+  return (
+    <>
+      <EuiFlexGrid columns={2} gutterSize="s">
+        <EuiFlexItem data-cy={`transaction-duration-charts`}>
+          <EuiPanel>
+            <React.Fragment>
+              <EuiFlexGroup justifyContent="spaceBetween">
+                <EuiFlexItem>
+                  <EuiTitle size="xs">
+                    <span>{responseTimeLabel(transactionType)}</span>
+                  </EuiTitle>
+                </EuiFlexItem>
+                <LicenseContext.Consumer>
+                  {(license) => (
+                    <MLHeader
+                      hasValidMlLicense={license?.getFeature('ml').isAvailable}
+                      mlJobId={charts.mlJobId}
+                    />
                   )}
-                />
-              </React.Fragment>
-            </EuiPanel>
-          </EuiFlexItem>
-
-          <EuiFlexItem style={{ flexShrink: 1 }}>
-            <EuiPanel>
-              <React.Fragment>
-                <EuiTitle size="xs">
-                  <span>{tpmLabel(transactionType)}</span>
-                </EuiTitle>
-                <TransactionLineChart
-                  series={tpmSeries}
-                  tickFormatY={this.getTPMFormatter}
-                  formatTooltipValue={this.getTPMTooltipFormatter}
-                  truncateLegends
-                />
-              </React.Fragment>
-            </EuiPanel>
-          </EuiFlexItem>
-        </EuiFlexGrid>
-        {transactionType === TRANSACTION_PAGE_LOAD && (
-          <>
-            <EuiSpacer size="s" />
-            <EuiFlexGrid columns={2} gutterSize="s">
-              <EuiFlexItem>
-                <EuiPanel>
-                  <DurationByCountryMap />
-                </EuiPanel>
-              </EuiFlexItem>
-              <EuiFlexItem>
-                <EuiPanel>
-                  <BrowserLineChart />
-                </EuiPanel>
-              </EuiFlexItem>
-            </EuiFlexGrid>
-          </>
-        )}
-      </>
-    );
-  }
+                </LicenseContext.Consumer>
+              </EuiFlexGroup>
+              <TransactionLineChart
+                series={responseTimeSeries}
+                tickFormatY={getResponseTimeTickFormatter(formatter)}
+                formatTooltipValue={getResponseTimeTooltipFormatter(formatter)}
+                onToggleLegend={onToggleLegend}
```

Comment:
WDYT about omitting the `onToggleLegend` function?

```suggestion
                onToggleLegend={setDisabledSeriesState}
```

Reply:
oh, thanks I totally missed that.